tools/stress-cluster/services/Stress.Watcher/src/JobEventHandler.cs

```diff
@@ -0,0 +1,237 @@
+        }
+
+        public bool ShouldDeleteResources(V1Job job)
+        {
+            if (!string.IsNullOrEmpty(Namespace) && Namespace != job.Namespace())
+            {
+                return false;
+            }
+
+            var initContainers = job.Spec?.Template?.Spec?.InitContainers;
+            if (initContainers == null || initContainers.Count() == 0) {
+                return false;
+            }
+
+            var deployContainers = initContainers.Where(c => c.Name == "init-azure-deployer");
+            if (deployContainers.Count() == 0)
+            {
+                return false;
+            }
+
+            /*
+               Check if job is done by determing whether any conditions are satifisfied as "True"
+               This is most flexible because checking succeeded/failed counts will not be flexible
+               when there are multiple pods specified for the job.
+
+               Example succeeded state
+               --------------------
+                 status:
+                   completionTime: "2022-10-26T20:53:20Z"
+                   conditions:
+                   - lastProbeTime: "2022-10-26T20:53:20Z"
+                     lastTransitionTime: "2022-10-26T20:53:20Z"
+                     status: "True"
+                     type: Complete
+                   ready: 0
+                   startTime: "2022-10-26T20:52:28Z"
+                   succeeded: 1
+               --------------------
+
+               Example failed state
+               --------------------
+                 status:
+                    conditions:
+                    - lastProbeTime: "2022-10-26T21:05:39Z"
+                      lastTransitionTime: "2022-10-26T21:05:39Z"
+                      message: Job has reached the specified backoff limit
+                      reason: BackoffLimitExceeded
+                      status: "True"
+                      type: Failed
+                    failed: 1
+                    ready: 0
+                    startTime: "2022-10-26T21:04:58Z"
+               --------------------
+           */
+            var isCompleted = job.Status?.Conditions?.Any(c => c.Status == "True");
+            if (isCompleted != true)
+            {
+                Logger.Information($"Job is not completed, skipping resource deletion.");
+            }
+
+            // NOTE: this will not handle labels added to a pod after deployment
+            job.Metadata.Labels.TryGetValue("Skip.RemoveTestResources", out var skipJobRemove);
+            job.Spec.Template.Metadata.Labels.TryGetValue("Skip.RemoveTestResources", out var skipPodRemove);
+
+            if (isCompleted == true && skipJobRemove == "true" || skipPodRemove == "true")
+            {
+                Logger.Information($"Resource has Skip.RemoveTestResources=true label, skipping resource deletion.");
+                return false;
+            }
+
+            return isCompleted == true;
```

Comment:
```suggestion
            return isCompleted;
```

Reply:
Heh, yeah it's ugly, but it's a nullable boolean type vs. just a boolean type, so I can't just return `isCompleted` since it's not a bool type. I guess I could change the function return type to nullable bool, but it felt simpler to contain it inside.

Reply:
makes sense to me